libs/utils/android/benchmark.py

```diff
@@ -0,0 +1,262 @@
+            self.bm_conf['device'] = self.args.android_device
+        if self.args.android_home:
+            self.bm_conf['ANDROID_HOME'] = self.args.android_home
+        if self.args.results_dir:
+            self.bm_conf['results_dir'] = self.args.results_dir
+        if self.args.collect:
+            self.bm_collect = self.args.collect
+
+        if self.args.iio_channel_map:
+            try:
+                em = self.bm_conf['emeter']
+                if em['instrument'] != 'acme':
+                    raise ValueError
+                em['channel_map'] = {}
+                for ch in self.args.iio_channel_map.split(','):
+                    ch_name, ch_id = ch.split(':')
+                    em['channel_map'][ch_name] = ch_id
+                self.bm_conf['emeter'] = em
+                self.log.info('Using iio-capture channels: %s', em)
+            except:
+                # No ACME configuration required by the test
+                pass
+
+        return self.bm_conf
+
+    def _getBmKind(self):
+        if self.bm_kind is None:
+            raise NotImplementedError("Override `bm_kind` attribute")
+        # Get a referench to the worload to run
+        self.wl = Workload(self.te).getInstance(self.bm_kind)
+        if self.wl is None:
+            raise ValueError("Specified benchmark [{}] is not supported"\
+                             .format(self.bm_kind))
+        return self.wl
+
+    def _getBmParams(self):
+        if self.bm_params is None:
+            raise NotImplementedError("Override `bm_params` attribute")
+        return self.bm_params
+
+    def _getBmCollect(self):
+        if self.bm_collect is None:
+            self.log.warning('No data collection configured')
+            return ''
+        return self.bm_collect
+
+    def __init__(self):
+        """
+        Set up logging and trigger running experiments
+        """
+        LisaLogging.setup()
+        self.log = logging.getLogger('Benchmark')
+
+        self.log.info('=== CommandLine parsing...')
+        self._parseCommandLine()
+
+        self.log.info('=== TestEnv setup...')
+        self.te = TestEnv(self._getBmConf())
+        self.target = self.te.target
+        # TODO: Check if TestEnv already provides support for rebooting
+        self._reboot()
+
+        self.log.info('=== Initialization...')
+        self._getBmKind()
+        self.out_dir=self.te.res_dir
+        if not self.benchmarkInit():
+            self.log.debug('Initialization failed: execution aborted')
+            return
+
+        self.log.info('=== Execution...')
+        self.wl.run(out_dir=self.out_dir,
+                    collect=self._getBmCollect(),
+                    **self.bm_params)
+
+        self.log.info('=== Finalization...')
+        self.benchmarkFinalize()
+
+    def _adb(self, cmd):
+        os.system('adb -s {} {}'.format(self.target.adb_name, cmd))
+
+    def _fastboot(self, cmd):
+        os.system('fastboot -s {} {}'.format(self.target.adb_name, cmd))
```

Comment:
but it's missing the device as a parameter... I can push a patch for devibe and get rid of both _adb and _fastboot locals defintions...